hack/devtools/rp_dev_helper.sh

```diff
@@ -1,39 +1,471 @@
-#!/bin/bash -e
+#!/bin/bash  
+
+set -o errexit \
+       -o nounset \
+       -o monitor
+
+declare -r utils="hack/util.sh"
+if [ -f "$utils" ]; then
+    source "$utils"
+fi
+
 ######## Helper file to run full RP dev either locally or using Azure DevOps Pipelines ########
+# Run usage_rp_dev to get functions' usage help
+secrets_dir="secrets"
+secrets_files=("dev-ca.crt" "dev-client.crt" "portal-client.pem" "firstparty.pem" "localhost.pem" "arm.pem"
+        "cluster-mdsd-self-signed.pem" "gwy-mdm-self-signed.pem" "gwy-mdsd-self-signed.pem" "rp-mdm-self-signed.pem"
+        "rp-mdsd-self-signed.pem" "full_rp_id_rsa" "full_rp_id_rsa.pub" "env" )
+
+    # # Define the expected directory and file names
+    # all_secret_files=("firstparty.key" "vpn-client.key" "vpn-eastus.ovpn" "localhost.crt" "proxy.crt"
+    #     "dev-ca.crt" "proxy-client.pem" "vpn-aks-eastus.ovpn" "portal-client.key" "portal-client.pem"
+    #     "proxy-client.key" "vpn-ca-cert.pem" "rp-metrics-int.pem" "dev-client.crt" "rp-logging-int.pem"
+    #     "vpn-ca.crt" "shared-cluster.kubeconfig" "vpn-client.pem" "firstparty.pem" "portal-client.crt"
+    #     "dev-ca.key" "vpn-aks-australiaeast.ovpn" "vpn-client-cert.pem" "vpn-client-key.pem" "dev-client.pem"
+    #     "proxy.key" "firstparty.crt" "vpn-client.crt" "vpn-aks-westeurope.ovpn" "localhost.key" "vpn-ca.pem"
+    #     "vpn-australiaeast.ovpn" "cluster-logging-int.pem" "vpn-ca.key" "localhost.pem" "proxy.pem" "dev-client.key"
+    #     "proxy_id_rsa.pub" "proxy_id_rsa" "vpn-ca-key.pem" "proxy-client.crt" "vpn-westeurope.ovpn" "dev-ca.pem"
+    #     "arm.crt" "arm.key" "arm.pem" "cluster-mdsd.crt" "cluster-mdsd.key" "cluster-mdsd.pem" "cluster-mdsd-self-signed.pem"
+    #     "gwy-mdm-self-signed.pem" "gwy-mdsd-self-signed.pem" "rp-mdm-self-signed.pem" "rp-mdsd-self-signed.pem"
+    #     "full_rp_id_rsa" "full_rp_id_rsa.pub" "env")
+
+verify_downloading_secrets() {
+    if [ ! -d "$secrets_dir" ]; then
+        abort "Directory '$secrets_dir' has not been created."
+    fi
+    #  TODO check if only the below files are required
+    for file in ${secrets_files[@]}; do
+        if [ ! -f "$secrets_dir/$file" ]; then
+            abort "File '$file' does not exist inside the directory '$secrets_dir'."
+        fi
+    done
+}
+
+check_jq_installed() {
+  if ! command -v jq &> /dev/null; then
+    abort "Error: jq is not installed.
+                Please install jq to proceed. You can install it using the following command:
+                For Debian/Ubuntu-based systems: sudo apt-get install jq
+                For Red Hat/CentOS-based systems: sudo yum install jq
+                For macOS: brew install jq"
+  fi
+}
+
+is_it_boolean(){
```

Comment:
Can we rename this to `is_boolean`?

```suggestion
is_boolean(){
```